lib/guardian/post_guardian.rb

```diff
@@ -173,8 +173,9 @@ def can_change_post_owner?
     is_admin?
   end
 
-  def can_wiki?
-    is_staff? || @user.has_trust_level?(TrustLevel[4])
+  def can_wiki?(post)
+    return false unless authenticated?
+    is_staff? || @user.has_trust_level?(TrustLevel[4]) || (@user.trust_level >= SiteSetting.min_trust_to_allow_self_wiki.to_i && is_my_own?(post))
```

Comment:
Instead of `(@user.trust_level >= SiteSetting.min_trust_to_allow_self_wiki.to_i)`, it might be possible to use `User#has_trust_level` here. :smile_cat: